Index: memcached-2.1.0/php_memcached.c
===================================================================
--- memcached-2.1.0.orig/php_memcached.c	2012-08-07 05:02:46.000000000 +0000
+++ memcached-2.1.0/php_memcached.c	2012-12-20 19:40:44.731728652 +0000
@@ -3870,6 +3870,12 @@
 	php_info_print_table_row(2, "Version", PHP_MEMCACHED_VERSION);
 	php_info_print_table_row(2, "libmemcached version", memcached_lib_version());
 
+#if HAVE_MEMCACHED_SASL
+  php_info_print_table_row(2, "SASL support", "yes");
+#else
+  php_info_print_table_row(2, "SASL support", "no");
+#endif
+
 #ifdef HAVE_MEMCACHED_SESSION
 	php_info_print_table_row(2, "Session support", "yes");
 #else
