Index: memcached-2.1.0/php_memcached.c
===================================================================
--- memcached-2.1.0.orig/php_memcached.c	2012-12-20 19:40:44.731728652 +0000
+++ memcached-2.1.0/php_memcached.c	2012-12-20 19:41:39.247728652 +0000
@@ -2642,11 +2642,19 @@
 	struct callbackContext* context = (struct callbackContext*) in_context;
 
 	hostport_len = spprintf(&hostport, 0, "%s:%d", memcached_server_name(instance), memcached_server_port(instance));
+#if defined(LIBMEMCACHED_VERSION_HEX) && LIBMEMCACHED_VERSION_HEX >= 0x01000008
 	version_len = snprintf(version, sizeof(version), "%d.%d.%d",
 				memcached_server_major_version(instance),
 				memcached_server_minor_version(instance),
 				memcached_server_micro_version(instance));
 
+#else
+  version_len = snprintf(version, sizeof(version), "%d.%d.%d",
+        instance->major_version,
+        instance->minor_version,
+        instance->micro_version);
+#endif
+
 	add_assoc_stringl_ex(context->return_value, hostport, hostport_len+1, version, version_len, 1);
 	efree(hostport);
 	return MEMCACHED_SUCCESS;
