Index: libmemcached-1.0.13/libhashkit/jenkins.cc
===================================================================
--- libmemcached-1.0.13.orig/libhashkit/jenkins.cc	2012-10-19 01:55:09.000000000 +0000
+++ libmemcached-1.0.13/libhashkit/jenkins.cc	2012-12-20 18:33:33.743728653 +0000
@@ -96,12 +96,14 @@
 uint32_t hashkit_jenkins(const char *key, size_t length, void *)
 {
   uint32_t a,b,c;                                          /* internal state */
-  union { const void *ptr; size_t i; } u;     /* needed for Mac Powerbook G4 */
+  #ifndef WORDS_BIGENDIAN
+    union { const void *ptr; size_t i; } u;
+    u.ptr = key;
+  #endif
 
   /* Set up the internal state */
   a = b = c = 0xdeadbeef + ((uint32_t)length) + JENKINS_INITVAL;
 
-  u.ptr = key;
 #ifndef WORDS_BIGENDIAN
   if ((u.i & 0x3) == 0)
   {
