diff -urN libmemcached-1.0.11/libhashkit/jenkins.cc libmemcached-1.0.11.new/libhashkit/jenkins.cc
--- libmemcached-1.0.11/libhashkit/jenkins.cc	2012-09-19 08:54:12.000000000 +0200
+++ libmemcached-1.0.11.new/libhashkit/jenkins.cc	2012-09-25 12:17:44.749995610 +0200
@@ -59,12 +59,14 @@
 uint32_t hashkit_jenkins(const char *key, size_t length, void *)
 {
   uint32_t a,b,c;                                          /* internal state */
-  union { const void *ptr; size_t i; } u;     /* needed for Mac Powerbook G4 */
+#ifndef WORDS_BIGENDIAN
+  union { const void *ptr; size_t i; } u;
+  u.ptr = key;
+#endif
 
   /* Set up the internal state */
   a = b = c = 0xdeadbeef + ((uint32_t)length) + JENKINS_INITVAL;
 
-  u.ptr = key;
 #ifndef WORDS_BIGENDIAN
   if ((u.i & 0x3) == 0)
   {
